File: Beecrowd/1068-Balanco-de-Parenteses/CPP/Main.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
	int esq = 0,
	    dir = 0;
	string str;

	cout << "\t0 - Encerrar programa" << endl;
	cin >> str;

	while (str != "0") {
		
		// Verifica de a palavra tem mais de zero caracteres e menos de dez mil caracteres
		if (str.length() >= 1 || str.length() <= 10000) {
			// Repetição passando por toda a palavra
			for (char letra : str) {
				// Verifica abertura de parênteses
				if (letra == '(') {
					esq++;
				}
				// Verifica fechamento de parênteses
				else if (letra == ')') {
					dir++;
				}
			}	

			// Verifica se abertura e fechamento são iguais
			if (esq == dir) {
				cout << "correct\n";
			}
			else {
				cout << "incorrect\n";
			}

			esq = 0; dir = 0; // Zerando variáveis
		}
		else {
			cout << "Frase inválida" << endl;
		}
		
		cin >> str; // Ler entrada
	}

	return 0;
}

File: Beecrowd/1068-Balanco-de-Parenteses/JAVA/Main.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		System.out.println("\t0 - Encerrar programa");

		String str = sc.nextLine();
		int dir = 0,
		    esq = 0;

		while (!str.equals("0")) {
			// Verifica se a palavra contém mais de zero caracteres e menos de dez mil caracteres
			if (str.length() >= 1 && str.length() <= 10000) {
				for (int i = 0; i < str.length(); i++) {
					// Verifica abertura de parênteses
				if (str.charAt(i) == '(') { 
					esq++;
				}
				// Verifica fechamento de parênteses
				else if (str.charAt(i) == ')') {
					dir++;
				}
			}

			// Verifica se abertura e fechamento são iguais
			if (esq == dir) {
				System.out.println("correct");
			}
			else {
				System.out.println("incorrect");
			}

			dir = 0; esq = 0; // Zera as variáveis inteiras

			str = sc.nextLine(); // Ler entrada
		}

		sc.close();
	}
}
